src/Microsoft.Graph.Core/Models/UploadSession.cs

```diff
@@ -26,5 +36,36 @@ internal class UploadSession : IUploadSession
         /// The URL for upload
         /// </summary>
         public string UploadUrl { get; set; }
+
+        /// <summary>
+        /// Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
+        /// </summary>
+        public IDictionary<string, object> AdditionalData { get; set; }
```

Comment:
```suggestion
        public IDictionary<string, object> AdditionalData { get; set; } = new Dictionary<string, object>();
```

and drop the constructor